### DISCUSSION TITLE
Issue DOTENV_ENV_VAR_RESOLUTION_CONFLICT while using env variables with useDotenv: true

### DISCUSSION THREAD

**Discussion by widedbenabd** [serverless/serverless#11156] [Q&A]
I have defined in my serverless file the parameter usedDotenv:true, and created some .env.dev/.env.test/.env.prod files. In those files I am declaring my variables as it should be and referring to those vars in the serverless file by env:STAGE

Now I am facing this issue, which I don't know how to solve **Cannot reliably resolve "env" variables due to resolution conflict.
  Environment variable "STAGE" which influences resolution of ".env" file were found to be defined in resolved ".env" file.DOTENV_ENV_VAR_RESOLUTION_CONFLICT**

Any ideas on the reason why this is appearing

**Comment by pgrzesik** [comment]
Hey @widedbenabd 👋 Could you please share a small reproducible example of your issue? Are you defining `stage` with an env var by any chance? If that's the case, then that is what is causing issues, because the stage needs to be resolved first in order to load stage-specific env file

**Reply by widedbenabd** [reply]
provider:
  name: aws
  lambdaHashingVersion: 20201221
  runtime: python3.7
  iam:
    role: testPythonLambda2
  tracing:
    lambda: true
  deploymentBucket:
    serverSideEncryption: AES256
  stage: ${env:STAGE, 'dev'}

this is how I am getting the stage in provider level. What other option should I have. It was fetching it from the options passed to the serverless command, but since that will be deprecated in serverless 3, we started fetching it from the env files.

**Reply by pgrzesik** [reply]
The best way is to pass it via `--stage <stage>` CLI flag, it will always take precedence over other settings. 

`stage: ${env:STAGE, 'dev'}` this only sets default stage and if someone passes `--stage` CLI flag it will be overriden anyway

**Reply by widedbenabd** [reply]
Alright I've put it back to opt:stage and it seems to be working now.

**Reply by widedbenabd** [reply]
Thanks @pgrzesik  for the quick reply

**Reply by pgrzesik** [reply]
If you are using `opt:stage` then you can remove setting `stage` in `provider.stage` altogether as internally the CLI option will take precedence anyway. Glad to hear you've been able to resolve the problem 👍
